From 24235060ac809d6e5bacfcc15e219aafd3af763f Mon Sep 17 00:00:00 2001
From: Mark Charlebois <charlebm@gmail.com>
Date: Fri, 29 Aug 2014 10:54:26 -0700
Subject: [PATCH] bcm_kona_smc

DO NOT UPSTREAM

This is a workaround for an illegal use of local register variables

Not-Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---
 arch/arm/mach-bcm/bcm_kona_smc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-bcm/bcm_kona_smc.c b/arch/arm/mach-bcm/bcm_kona_smc.c
index a55a7ec..ae89732c 100644
--- a/arch/arm/mach-bcm/bcm_kona_smc.c
+++ b/arch/arm/mach-bcm/bcm_kona_smc.c
@@ -20,7 +20,6 @@
 #include <linux/of_address.h>
 
 #include "bcm_kona_smc.h"
-
 static u32		bcm_smc_buffer_phys;	/* physical address */
 static void __iomem	*bcm_smc_buffer;	/* virtual address */
 
@@ -119,12 +118,14 @@ static int bcm_kona_do_smc(u32 service_id, u32 buffer_phys)
 	r6 = buffer_phys;
 
 	asm volatile (
+#if 0
 		/* Make sure we got the registers we want */
 		__asmeq("%0", "ip")
 		__asmeq("%1", "r0")
 		__asmeq("%2", "r4")
 		__asmeq("%3", "r5")
 		__asmeq("%4", "r6")
+#endif 
 #ifdef REQUIRES_SEC
 		".arch_extension sec\n"
 #endif
-- 
1.9.1

